### DISCUSSION TITLE
Change dev-file db vendor

### DISCUSSION THREAD

**Discussion by DGuhr** [keycloak/keycloak#10295] [Archive]
At the moment, the standard "try-out keycloak" database is h2-file, in version 1.4.197

Due to multiple CVEs in the past, I want to bring up the discussion to migrate from h2 to other database vendors like apache derby, or to our new ConcurrentHashMap based storage (filebased). First step in that direction was to rename the vendor in the new Keycloak from `h2-file` to `dev-file`, so a change of the underlying db would not be a breaking change for the configuration. 

As stated in this [discussion](https://github.com/keycloak/keycloak/pull/10206#discussion_r808737032) on a slightly related PR, there are multiple things to consider:
* arquillian testsuite uses this database.
* dev-mode uses this database.
* IDELauncher uses this database, bc. it starts using start-dev
* map based persistence wouldn't have constraints other db vendors have, which is not necessarily a good thing.

I have no clear solution, and I don't think I have all the context yet, but wanted to bring this up to see what you people think. So please don't hold back with feedback :)

**Comment by stianst** [comment]
One issue with the map store is that federation like user federation is done differently, so that may not be an option until we switch to the new store as the default.

In the long run we would want to switch to the static store, but that is not ready yet for sure.

So, perhaps for now we're stuck with h2, unless we consider Derby as a stop-gap solution.
